pjsip/src/pjsua-lib/pjsua_core.c

```diff
@@ -660,6 +660,38 @@ static pj_bool_t mod_pjsua_on_rx_request(pjsip_rx_data *rdata)
 {
     pj_bool_t processed = PJ_FALSE;
 
+    if (pjsip_tsx_detect_merged_requests(rdata)) {
+        pjsip_transaction *new_tsx;
+        pjsip_tx_data *tdata;
+        pj_status_t status;
+
+        PJ_LOG(4, (THIS_FILE, "Merged request detected"));
+
+        /* Create new tsx and answer with 482 (Loop Detected) */
+        status = pjsip_tsx_create_uas(&pjsua_var.mod, rdata,
+                                      &new_tsx);
```

Comment:
The standard mentions about passing it to server transaction indeed, but not really sure what that is for, as the new tsx will be terminated immediately below (PJSUA call usually responds statelessly/without-tsx).

Reply:
The reason is probably because of retransmission and such. The problem is I don't know who's going to be responsible to make sure that the transaction is going to be terminated and will not leak, so I just terminate it immediately.

Or should we change this to `pjsip_endpt_send_response()` instead?

Reply:
Ah right, retransmission etc.

If you mean `pjsip_endpt_respond()` (respond statefully), perhaps yes, for a bit simpler code.